src/ai/backend/manager/models/vfolder.py

```diff
@@ -1779,6 +1784,35 @@ async def delete_vfolders(
         )
 
 
+async def delete_vfolders(
+    requested_vfolders: Sequence[VFolderDeletionInfo],
+    *,
+    storage_manager: StorageSessionManager,
+    db: ExtendedAsyncSAEngine,
+    reporter: ProgressReporter | None = None,
+) -> None:
+    """
+    Spawn vfolder deletion tasks grouped by a name of storage proxies.
+    """
+
+    def _keyfunc(item: VFolderDeletionInfo) -> str:
+        proxy_name, _ = storage_manager.split_host(item.host)
+        return proxy_name
+
+    async with aiotools.TaskGroup() as tg:
+        for _, vfolder_iterator in itertools.groupby(
+            sorted(requested_vfolders, key=_keyfunc), key=_keyfunc
```

Comment:
In this case, using `typing.Iterable` seems more suitable than `typing.Sequence` for `requested_vfolders`.
(ref: https://docs.python.org/3/library/functions.html#sorted)